pkg/agent/git/service.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package git
+
+import (
+	"bytes"
+	"os"
+	"path"
+
+	"github.com/daytonaio/daytona/pkg/agent/config"
+	"github.com/daytonaio/daytona/pkg/serverapiclient"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"gopkg.in/ini.v1"
+)
+
+type Service struct {
+	Config *config.Config
```

Comment:
I would avoid sending the entire config down, especially as a pointer.

I suggest only passing `ProjectDir string` to the service